xdem/volume.py

```diff
@@ -0,0 +1,49 @@
+from __future__ import annotations
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+
+
+def hypsometric_binning(ddem: np.ndarray, dem: np.ndarray):
+
+    assert ddem.shape == dem.shape
+    nan_mask = np.isnan(ddem) | np.isnan(dem)
+    ddem = ddem[~nan_mask]
+    dem = dem[~nan_mask]
+
+    old_dem = dem - ddem
+    elevation_range = np.percentile(old_dem, 99) - np.percentile(old_dem, 1)
+
+    bin_size = 50 if elevation_range > 500 else elevation_range / 10
+
+    bins = np.arange(old_dem.min(), old_dem.max() + bin_size, step=bin_size)
+    indices = np.digitize(old_dem, bins=bins)
+
+    medians = np.zeros(shape=bins.shape[0] - 1, dtype=ddem.dtype) + np.nan
```

Comment:
to initialize those in one line, I usually write:
`medians, means, stds = ( np.zeros(shape=,dtype=) * np.nan for i in range(3) )`

Reply:
Or simply:
`medians, means, stds = ( np.zeros(shape=,dtype=) * np.nan) * 3`

Reply:
Missed a comma:
`medians, means, stds = ( np.zeros(shape=,dtype=) * np.nan, ) * 3`

Reply:
Implemented @adehecq 's syntax in 5853875.

Reply:
@adehecq I knew something didn't work with this syntax (tried this in the past)! Using `*3` creates linked objects, so if for instance you do medians[0] = 1 it will change the value of means[0] and stds[0] into 1 as well.
```
medians, means, stds = ( np.zeros(3) * np.nan, ) * 3
means[0] = 1
stds[0]
Out[9]: 1.0
```
@erikmannerfelt Didn't you encounter that issue when updating the syntax? I need to correct this in spstats as well (I was updating similarly on my side)

Reply:
Yes, good to know!!